x/distribution/keeper/delegation.go

```diff
@@ -247,15 +249,44 @@ func (k Keeper) withdrawDelegationRewards(ctx context.Context, val stakingtypes.
 	finalRewards, remainder := rewards.TruncateDecimal()
 
 	// add coins to user account
+	// or put them in pending rewards
 	if !finalRewards.IsZero() {
-		withdrawAddr, err := k.GetDelegatorWithdrawAddr(ctx, delAddr)
-		if err != nil {
+		key := collections.Join(sdk.AccAddress(delAddr), sdk.ValAddress(valAddr))
+		// check if the user has pending rewards
+		// we need to check the pending rewards, in case a user has
+		// redelegated away and redelegated back to the same validator
+		// or to set them if withdrawNow is false
+		outstanding, err := k.UserOutstandingRewards.Get(ctx, key)
```

Comment:
[nit] in this PR outstanding and pending are used interchangeably. Can we standardize on one term please?

Reply:
updated!